packages/property-grid/src/components/PresentationPropertyGridWidget2.tsx

```diff
@@ -0,0 +1,736 @@
+  );
+
+  const onPropertyContextMenu = React.useCallback(
+    (args: PropertyGridContextMenuArgs) => {
+      args.event.persist();
+      setupContextMenu(args);
+    },
+    [setupContextMenu]
+  );
+
+  const onContextMenuOutsideClick = React.useCallback(() => {
+    setContextMenu(undefined);
+  }, []);
+
+  const onContextMenuEsc = React.useCallback(() => {
+    setContextMenu(undefined);
+  }, []);
+
+  const favoriteActionButtonRenderer = React.useCallback(
+    (props: ActionButtonRendererProps) => {
+      if (iModelConnection && dataProvider) {
+        const { property } = props;
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        const field = useAsyncValue(
+          // eslint-disable-next-line react-hooks/rules-of-hooks
+          React.useMemo(
+            async () => dataProvider.getFieldByPropertyRecord(property),
+            [property]
+          )
+        );
+
+        return (
+          <div>
+            {field &&
+              (Presentation.favoriteProperties.has(
+                field,
+                iModelConnection,
+                FavoritePropertiesScope.IModel
+              ) ||
+                props.isPropertyHovered) && (
+                <FavoriteActionButton field={field} imodel={iModelConnection} />
+              )}
+          </div>
+        );
+      }
+      return null;
+    },
+    [dataProvider, iModelConnection]
+  );
+
+  const renderHeader = () => {
+    return (
+      <div className="property-grid-react-panel-header">
+        {props.onBackButton !== undefined && (
+          <div
+            className="property-grid-react-panel-back-btn"
+            onClick={props.onBackButton}
+          >
+            <Icon
+              className="property-grid-react-panel-icon"
+              iconSpec="icon-progress-backward"
+            />
+          </div>
+        )}
+        <div className="property-grid-react-panel-label-and-class">
+          <div className="property-grid-react-panel-label">
+            {title && PropertyValueRendererManager.defaultManager.render(title)}
+          </div>
+          <div className="property-grid-react-panel-class">{className}</div>
+        </div>
+        {props.onInfoButton !== undefined && (
+          <div
+            className="property-grid-react-panel-info-btn"
+            onClick={props.onInfoButton}
+          >
+            <Icon
+              className="property-grid-react-panel-icon"
+              iconSpec="icon-info-hollow"
+            />
+          </div>
+        )}
+      </div>
+    );
+  };
+
+  const renderPropertyGrid = () => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { isOverLimit } = usePropertyDataProviderWithUnifiedSelection({
+      dataProvider: dataProvider as IPresentationPropertyDataProvider,
+    });
+
+    if (isOverLimit) {
+      return (
+        <FillCentered>
+          {IModelApp.i18n.translate(
+            "uiTestExtension:properties.too-many-elements-selected"
```

Comment:
?? 
left over from bills code?
i dont think we care if its overlimit, that what the virtualized property grid does, or so i thought

Reply:
@aruniverse I think we need to check with @grigasp on this as the hook still provides this property.

Reply:
The docs for this property say: [Whether selected element count is exceeding the limit](https://github.com/imodeljs/imodeljs/blob/27d9a7a33ba5a6c4e0c96a5246003381b0ff7d2e/presentation/components/src/presentation-components/propertygrid/UseUnifiedSelection.tsx#L44). 

Requesting properties for a very large number of elements (e.g. after fence selecting thousands or millions of elements) may be very expensive and in return the result is pretty much useless. For that we added this feature to avoid loading content for when requesting it for "too many" elements. "Too many" can be configured with [requestedContentInstancesLimit prop](https://github.com/imodeljs/imodeljs/blob/27d9a7a33ba5a6c4e0c96a5246003381b0ff7d2e/presentation/components/src/presentation-components/propertygrid/UseUnifiedSelection.tsx#L33) and defaults to `100`. When the number of elements we're requesting content for exceeds this, the `isOverLimit` flag is set to allow the widget to emit a useful message to the users. Otherwise they'd simply see an empty widget content.

I'm not sure if there's a case where the widget is displayed when nothing is selected (content is requested for 0 elements), but if it is possible, you may also want to add a similar check to [this](https://github.com/imodeljs/imodeljs/blob/27d9a7a33ba5a6c4e0c96a5246003381b0ff7d2e/test-apps/presentation-test-app/src/frontend/components/properties-widget/PropertiesWidget.tsx#L162-L164).

HTH